bin/pyenv-installer

```diff
@@ -50,15 +52,78 @@ checkout "${GITHUB}/pyenv/pyenv-update.git"     "${PYENV_ROOT}/plugins/pyenv-upd
 checkout "${GITHUB}/pyenv/pyenv-virtualenv.git" "${PYENV_ROOT}/plugins/pyenv-virtualenv"  "master"
 checkout "${GITHUB}/pyenv/pyenv-which-ext.git"  "${PYENV_ROOT}/plugins/pyenv-which-ext"   "master"
 
-if ! command -v pyenv 1>/dev/null; then
-  { echo
-    colorize 1 "WARNING"
-    echo ": seems you still have not added 'pyenv' to the load path."
-    echo
+write_source() {
+  {
+    echo "Appending the following lines to $1:"
+    echo '  # pyenv'
+    echo '  export PATH="'"$PYENV_ROOT/bin:"'$PATH"'
+    echo '  eval "$(pyenv init --path)"'
+    echo '  eval "$(pyenv virtualenv-init -)"'
+    echo ''
   } >&2
 
-  { # Without args, `init` commands print installation help
-    "${PYENV_ROOT}/bin/pyenv" init || true
-    "${PYENV_ROOT}/bin/pyenv" virtualenv-init || true
-  } >&2
+  echo "" >>$1
+  echo '# pyenv' >>$1
+  echo 'export PATH="'"$PYENV_ROOT/bin:"'$PATH"' >>$1
+  echo 'eval "$(pyenv init --path)"' >>$1
+  echo 'eval "$(pyenv virtualenv-init -)"' >>$1
+
+  SRC_WRITTEN=true
+}
+
+add_userpath() {
+  OS="$(uname -s)"
+  CURRENT_SHELL="$(basename "$SHELL")"
```

Comment:
Since this is a Bash script, `$SHELL` would probably always be Bash.
`pyenv-init` has logic to detect the shell of the parent process.
That said, this logic seems to rather belong in `pyenv-init` to avoid duplication. E.g. it likewise determines profile files for diffent shells and commands to write -- we just need to actually write them rather than just display.

Reply:
[pyenv-init]: https://github.com/pyenv/pyenv/blob/master/libexec/pyenv-init
@native-api,  

After inspecting [pyenv-init], I think we should just copy the logic of determining user shell directly from it, because: 

1. The script(`pyenv-init`) is not composed to be sourced by other one, so that we can't simply source it and variables(`rc`, `shell`, `profile`) in our script.
2. `bin/py-installer` should be a self-containing script, which could be runned even if `pyenv-init` changed, so a little duplication might be acceptable.

So here's what I'll try to do next:
+ Copying the logic directly of determining shell & rc files from [pyenv-init] and put a reference to the original code.
+ Keep the `OS` variable so that it can better fit mac users.

I'll start working on this if you think it's appropriate

Reply:
The idea is to add yet another mode to `pyenv-init` -- e.g. `--install` -- that would write shell configuration. Just like currently, the installer runs it without arguments and it prints installation instructions.

Apart from avoiding code duplication, we won't need to adapt Pyenv-installer if we change something in the suggested shell configuration.

Reply:
I copied the shell detection mechanism from `pyenv-init` in `pyenv-installer`.

This resolves the issue of `pyenv-init` being unable to detect the parent of the parent process when invoked through `pyenv-installer`, as described in my experiment(https://github.com/ianchen-tw/pyenv-installer/issues/2).